src/parquet/src/optimizers/lineage_opt.rs

```diff
@@ -1034,10 +1076,19 @@ mod tests {
         let expected_field_metadata = expected
             .iter()
             .map(|extraction| {
-                (
-                    extraction.column.name().to_string(),
-                    extraction.component.metadata_value().to_string(),
-                )
+                let mut sorted_units: Vec<&SupportedIntervalUnit> =
+                    extraction.components.iter().collect();
+                sorted_units.sort_by_key(|unit| match unit {
+                    SupportedIntervalUnit::Year => 0,
+                    SupportedIntervalUnit::Month => 1,
+                    SupportedIntervalUnit::Day => 2,
+                });
+                let metadata_value = sorted_units
+                    .iter()
+                    .map(|unit| unit.metadata_value())
+                    .collect::<Vec<_>>()
+                    .join(",");
```

Comment:
I'm not sure joining them with `,` is the best approach, but I believe you'll make it work when integrating this into cache storage.